prov/efa/src/efa_hmem.c

```diff
@@ -178,52 +223,90 @@ static int efa_hmem_support_status_update_synapseai(struct efa_hmem_support_stat
 		return 0;
 	}
 
-	synapseai_status->initialized = true;
-	synapseai_status->p2p_supported = true;
+	synapseai_info->initialized = true;
+	synapseai_info->p2p_supported = true;
+
+	/*  Only the long read protocol is supported */
+	if (-FI_ENODATA != fi_param_get(&rxr_prov, "inter_max_medium_message_size", &tmp_value) ||
+		-FI_ENODATA != fi_param_get(&rxr_prov, "inter_min_read_message_size", &tmp_value) ||
+		-FI_ENODATA != fi_param_get(&rxr_prov, "runt_size", &tmp_value)) {
+		EFA_WARN(FI_LOG_DOMAIN,
+		         "One of the following environment variable(s) was set: (FI_EFA_INTER_MAX_MEDIUM_MESSAGE_SIZE, "
+		         "FI_EFA_INTER_MIN_READ_MESSAGE_SIZE, and/or FI_EFA_RUNT_SIZE), but EFA HMEM via Synapse API "
+		         "only supports long read protocol.  Please unset the env variable(s) and re-run.\n");
+	}
+
+	synapseai_info->max_medium_msg_size = 0;
+	synapseai_info->min_read_msg_size = 0;
```

Comment:
the value should be 1. we still want to use eager protocol for 0 sized message.

Reply:
done